src/odemis/acq/stream/_sync.py

```diff
@@ -136,13 +136,22 @@ def __init__(self, name, streams):
                 break
 
         # Get polarization analyzer if found in optical substream
+        self._analyzer = None
         for s in streams:
             # check if stream has attribute analyzer and analyzer is not None
             if hasattr(s, "analyzer") and s.analyzer:
-                # get polarization analyzer
-                self._analyzer = s.analyzer
-                # get polarization analyzer VA
-                self._polarization = s.polarization
+                # additionally check that not multiple streams have a component analyzer
+                if not self._analyzer:
+                    # get polarization analyzer
+                    self._analyzer = s.analyzer
+                    # get polarization analyzer VA
+                    self._polarization = s.polarization
+                else:
+                    raise Exception("Found multiple streams with a component polarization analyzer!"
+                                    "Can have only one stream with a component polarization analyzer.")
+
+        # optical detector is always last stream in streams
+        self._det_num = -1  # optical detector is always last in streams
```

Comment:
This can only go to the `SEMCCDMDStream.__init__()`. I'd suggest to name it `._ccd_num` to follow the other variables name. Also, to ensure it works really in all cases, avoid `-1` but set it to `len(self._streams) - 1`.